src/js/editor/edit.js

```diff
@@ -54,19 +55,21 @@ export default function ( { className, attributes, setAttributes } ) {
 		className += ' has-complete-msg';
 	}
 
+	setAttributes( {postId: select( 'core/editor' ).getCurrentPostId()} );
```

Comment:
My experience with these data calls is that on page load, if the block exists in the editor, it'll make the calls when the block initializes and the postId may return null, to avoid multiple unnecessary sets with empty data we can check for the presence of the id and only set it when it exists

Also, once it's been set the postId isn't going to change so we should be able to make this a one-time thing.

```suggestion
	let { postId } = attributes;
	postId = postId || select( 'core/editor' ).getCurrentPostId();
	if ( postId && ! attributes.postId ) {
		setAttributes( { postId } );
	}
```

Reply:
I didn't test this.

Reply:
Ok for not setting it when null, BUT I wanted it to be set not only once because I was thinking about copy pasting a block in another page, and in that case I wanted it to get the "new page" id. Although I'm going to test if your solution works fine in this case <3 thanks!

Reply:
Ok @thomasplevy just tested that with your code above if I copy this block into another page and save then the block retains the previous post-id attribute... So maybe we set it only if the current post id is not null and different from the saved attribute... agreed?

Reply:
hadn't thought about that, good call